instrumentation/dalli/lib/opentelemetry/instrumentation/dalli/patches/client.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Dalli
+      module Patches
+        # Module to prepend to Dalli::Client for instrumentation
+        module Client
```

Comment:
Haven't used Dalli in production before, could you explain a bit the benefit of instrumenting the higher level `Dalli::Client.perform` and `Dalli::Client.get_multi_yielder`, as opposed to `Dalli::Server.request`?

Basically, by choosing to wrap `Dalli::Client.get_multi_yielder` we introduce a bit of a tradeoff it seems, where we're losing some visibility into the underlying `host/port` info of the server, since `get_multi_yielder` calls `server.request` directly without going thru perform, if i'm reading this correctly? https://github.com/petergoldstein/dalli/blob/ca8b286de230546c3bfe2789e9b6bc5a2c619570/lib/dalli/client.rb#L429

So in those cases we'd lose the  `net.peer.name` and `net.peer.port` attributes, (since capturing it for a query that hits multiple servers introduces a bit of a race condition), and if there are errors we may not be able to attribute it to a specific server. But, as a tradeoff, we get better visibility into the timing of the entire operation, which is what the user is performing and likely cares more about. Am I stating that correctly? And I guess, more importantly is this intentional and something you feel is a greater value add? I defer to your opinion here but think it's worth being explicit about.

My other thought here is maybe we wrap both? IE, wrap the multi yielder, but also generate child spans for each `server.request`. That way if say, there's a connection issue or latency on a specific server, it's clear which one, while with the `get_multi` parent span we also _get_ (_hehe_) the visibility into what the broader user query was? Not sure if that's worth the implementation effort but just throwing it out there

Reply:
Thanks for the suggestion. When writing our instrumentation, we saw `Client.perform` as the "Chokepoint method for instrumentation", and I guess we didn't dig deeper. Looking at `Server.request`, I see that's also marked as "Chokepoint method for instrumentation", and it is likely better to instrument at that level, closer to the wire. I think we should instrument *only* at that level, rather than introducing additional spans at the higher level.